modules/dqt/ajax/datadictionary.php

```diff
@@ -43,18 +43,23 @@
             "endkey"   => "[\"$category\", \"ZZZZZZZZ\"]",
         ]
     );
-} else if (isset($_REQUEST['key']) && $_REQUEST['key']) {
-    $key = explode('%2C', urlencode($_REQUEST['key']));
+} else if (isset($_REQUEST['keys']) && $_REQUEST['keys']) {
+    $keys    = json_decode($_REQUEST['keys']);
+    $results = [];
+    foreach (array_keys($keys) as $index) {
+        $key = explode('%2C', urlencode($keys[$index]));
 
-    $results = $cdb->queryView(
-        "DQG-2.0",
-        "datadictionary",
-        [
-            "reduce" => "false",
-            "key"    => "[\"$key[0]\",\"$key[1]\"]",
-        ]
-    );
+        $query = $cdb->queryView(
+            "DQG-2.0",
+            "datadictionary",
+            [
+                "reduce" => "false",
+                "key"    => "[\"$key[0]\",\"$key[1]\"]",
+            ]
+        );
+
+        array_push($results, $query[0]);
```

Comment:
just wanna check that it should be $query[0] instead or just $query, which was how the results were being used in the deleted code on the left here for the case where there's only 1 key

Reply:
Hi @zaliqarosli, please manually test if you're concerned because that change is from the logic change for fixing the performance problem. No data is lost that I've aware about and the filter & fields will load for the DQT to function from the tests I did.

Reply:
Hi @zaliqarosli & @pierre-p-s, I went and took a few screen shots of before and after to help illustrate what I did in the past. Basically, I cleaned up the data being sent back to the frontend and also note that the frontend was modified to not require many requests but just one. So the $query[0] is part of the improvement.

Before:
![before](https://user-images.githubusercontent.com/16293415/100133917-6403ca80-2e55-11eb-91e7-a98385e336cd.jpg)


After:
![after](https://user-images.githubusercontent.com/16293415/100133925-66febb00-2e55-11eb-91cd-a9851467fd5f.jpg)